app/jobs/update_consultee_email_status_job.rb

```diff
@@ -30,11 +30,20 @@ def perform(consultee_email)
       current_time = Time.current
 
       if consultee_email.delivered?
-        consultee.update!(
-          status: "awaiting_response",
-          email_delivered_at: consultee.email_delivered_at || current_time,
-          last_email_delivered_at: current_time
-        )
+        consultee.with_lock do
+          last_resp_at = consultee.last_response_at
+
+          attrs = {
+            email_delivered_at: consultee.email_delivered_at || current_time,
+            last_email_delivered_at: current_time
+          }
+
+          if last_resp_at.nil? || last_resp_at < current_time
+            attrs[:status] = :awaiting_response
+          end
```

Comment:
Can last_response_at ever be in the future?

Reply:
That's the race condition I believe. I think Tom managed to add a response quick enough that it did seem to cause to revert back to awaiting_response here.

Reply:
I'm trying to understand how that could happen. It's possible that if we were taking the date of an email, for example, and trusting that, it would be possible to set the email's sent date to the future and we'd need to account for that.

But in this case it sounds like the dates are both being set from our system and so I don't see how it can be possible for a response to arrive in the future.

(i.e., `last_resp_at == current_time` at the instant the response is created; so `last_resp_at` should by definition be equal to or greater than `current_time`.)